packages/app/src/components/catalog/ClusterStatus.tsx

```diff
@@ -0,0 +1,152 @@
+  name.replace(/[A-Z]/g, v => (` ${v.toLowerCase()}`))
+    .replace(/^\w/g, v => v.toUpperCase())
+)
+
+const valueFormatter = (value: any): any => {
+  if (typeof value === 'string') {
+    if (value.slice(-2) === 'Ki') {
+      return convertToGibibytes(value)
+    } else if (value.slice(0, 4) === 'http') {
+      return <Link to={value}>{value}</Link>
+    }
+  }
+  return value.toString()
+}
+
+const keyExists = (object: object, key: string): boolean => (
+  Object.keys(object).includes(key) ? true : false
+)
+
+const tableFromObject = (data: any, title: string,): any => {
+  const values = Object.entries(data)
+    .reduce((filtered: any, entry) => {
+      if (typeof entry[1] !== 'object') {
+        filtered.push(
+          {
+            name: nameFormatter(entry[0]),
+            value: valueFormatter(entry[1])
+          }
+        );
+      }
+      return filtered;
+    }, []);
+
+  if (values.length === 0) {
+    return <></>
+  }
+  return (
+    <Card>
+      <CardHeader
+        title={title}
+      />
+      <CardContent style={{ padding: 0 }}>
+        <Table
+          options={{
+            search: false,
+            paging: false,
+            toolbar: false,
+            header: false,
+            padding: 'dense',
+          }}
+          data={values}
+          columns={[
+            { field: 'name', highlight: true, width: '15%', cellStyle: { whiteSpace: 'nowrap' } },
+            { field: 'value' },
+          ]}
+        />
+      </CardContent>
+    </Card>
+  )
+}
+
+export const ClusterInfoTable = ({ cluster }: props) => (
+  tableFromObject(cluster, 'Cluster details')
+)
+
+export const ClusterAvaliableResourceTable = ({ cluster }: props): any => (
+  keyExists(cluster, 'availableResources')
+    ? tableFromObject(cluster.allocatableResources, 'Available')
+    : <></>
+)
+
+export const ClusterAllocatableResourceTable = ({ cluster }: props): any => (
+  keyExists(cluster, 'allocatableResources')
+    ? tableFromObject(cluster.allocatableResources, 'Allocatable')
+    : <></>
+)
+
+export const ClusterStatusCard = ({ cluster }: props): any => (
+  <InfoCard title="Status" divider={false} >
+    <div style={{ textAlign: 'center', margin: 0 }}>
+      <Typography variant="h1">
+        {cluster.status.available ? <StatusOK /> : <StatusError />}
+      </Typography>
+      <Typography variant="subtitle1">
+        {cluster.status.reason}
+      </Typography>
+    </div>
+  </InfoCard>
+)
+
+export const ClusterStatus = () => {
+  const { entity } = useEntity();
+  const configApi = useApi(configApiRef);
+  const [cluster, setCluster] = useState<ClusterDetails>({
+    status: {
+      available: false,
+      reason: 'Loading'
+    }
+  });
+  const [_, refresh] = useAsyncFn(
+    async () => {
+      const thisCluster = await getClusterByName(configApi, entity.metadata.name)
+      setCluster(thisCluster)
+    }, [], { loading: true }
+  );
+  useDebounce(refresh, 10);
```

Comment:
This would be a nice context provider. Instead of passing `cluster` prop everywhere, let it consume a context hook `useCluster()`. The highlighted section should be the context provider.

Reply:
Updated in the latest push.